scripts/apis_scan.py

```diff
@@ -37,6 +39,21 @@ def create_session(session_name):
 
 
 def enable_httpsender_script():
+
+    if config.HTTP_SENDER_SCRIPT_PARAMS != "":
+        # Set Cookie var from config 
+        template_env = jinja2.sandbox.SandboxedEnvironment(
```

Comment:
Is a jinja template needed ? I feel like there might be easier simpler solution

Reply:
This is how I found I was able to get the variable from the config file and rewrite the variables into the authentication script. I had found this in another part of the code. If you know of a different solution I'd definitely be open. Just used what I found (not familiar with that package anyways)

Reply:
While trying the PR, I got this error :

```
jdk.nashorn.internal.runtime.ParserException: <eval>:40:14 Expected ident but found {
var params = {{ params | tojson | safe }}; // pass a dictionary wtih cookieName and cookieVal as keys
              ^No script interface found for HttpSender script
```
(and no cookie were sent to the server, but I am not sure if this is related) 

As far as I understand the patch : the `scripts/add-token-cookie-param.js` file is actually not a Javascript file, but a Jinja template that produces a Javascript file, replacing `{{ params | tojson | safe }}` by the actual value. 
However, ZAP still loads the `add-token-cookie-param.js` "template" as a script, because it is in the script directory, and fails while parsing it.

IIRC, ZAP has a way to send parameters to HttpSender scripts, I will check this.

Reply:
Here's a method that works (I just tried), with minimal change from the original patch:

in the JS, set params as such :  

```
// cookieVal will be replaced with a token that can be obtained via 'Copy login command' menu in the console
var params = {
        cookieName: org.zaproxy.zap.extension.script.ScriptVars.getScriptVar(this.context, "cookieName"),
        cookieVal: org.zaproxy.zap.extension.script.ScriptVars.getScriptVar(this.context, "cookieVal")
};
```

In the api_scan, configure the cookie as such (after it is loaded! : you will have to move it towards the end of the `enable_httpsender_script()` function, just before it returns):

```python
    # If the script requires cookie parameter
    if config.HTTP_SENDER_SCRIPT_PARAMS != "":
        for k, v in config.HTTP_SENDER_SCRIPT_PARAMS.items():
            ret = script.set_script_var(config.HTTP_SENDER_SCRIPT_NAME, k, v)
            logging.info(f"httpsenderscript: set Variable '{k}' to value '{v}', returned: {ret}")
```

:) feel free to improve on this.

(also: the additional imports can be removed)

Reply:
awesome!!! thanks so much for that Cedric. Will give that a try today and update the PR.